// spinners-react/lib/cjs/withSharedProps-EQOgeSQG.js
'use strict';

var React = require('react');

/******************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */
/* global Reflect, Promise, SuppressedError, Symbol, Iterator */


exports.__assign = function() {
    exports.__assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return exports.__assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

typeof SuppressedError === "function" ? SuppressedError : function (error, suppressed, message) {
    var e = new Error(message);
    return e.name = "SuppressedError", e.error = error, e.suppressed = suppressed, e;
};

var defaultProps = {
    color: '#38ad48',
    enabled: true,
    size: 50,
    style: {},
};
var normalizeSize = function (size) { return (parseFloat(size.toString()).toString() === size.toString()
    ? "".concat(size, "px")
    : size.toString()); };
var withSharedProps = function (Component) {
    function Wrapper(props) {
        var _a = props.color, color = _a === void 0 ? defaultProps.color : _a, _b = props.enabled, enabled = _b === void 0 ? defaultProps.enabled : _b, _c = props.size, size = _c === void 0 ? defaultProps.size : _c, _d = props.style, style = _d === void 0 ? defaultProps.style : _d, otherProps = __rest(props, ["color", "enabled", "size", "style"]);
        var componentProps = exports.__assign(exports.__assign({}, otherProps), { style: exports.__assign({ color: color, overflow: 'visible', width: normalizeSize(size) }, style) });
        if (!enabled)
            return null;
        return React.createElement(Component, exports.__assign({}, componentProps));
    }
    return Wrapper;
};

exports.__rest = __rest;
exports.withSharedProps = withSharedProps;
